framework/base/Controller.php

```diff
@@ -77,13 +77,13 @@ class Controller extends Component implements ViewContextInterface
     public $response = 'response';
 
     /**
-     * @var View the view object that can be used to render views or view files.
+     * @var View|null the view object that can be used to render views or view files.
      */
-    private $_view;
+    private $_view = null;
     /**
-     * @var string the root directory that contains view files for this controller.
+     * @var string|null the root directory that contains view files for this controller.
      */
-    private $_viewPath;
+    private $_viewPath = null;
```

Comment:
```php
/**
     * Returns the directory containing view files for this controller.
     * The default implementation returns the directory named as controller [[id]] under the [[module]]'s
     * [[viewPath]] directory.
     * @return string the directory containing the view files for this controller.
     */
    public function getViewPath()
    {
        if ($this->_viewPath === null) {
            $this->_viewPath = $this->module->getViewPath() . DIRECTORY_SEPARATOR . $this->id;
        }

        return $this->_viewPath;
    }

    /**
     * Sets the directory that contains the view files.
     * @param string $path the root directory of view files.
     * @throws InvalidArgumentException if the directory is invalid
     * @since 2.0.7
     */
    public function setViewPath($path)
    {
        $this->_viewPath = Yii::getAlias($path);
    }
```

This code use `_viewPath` and  there is necessary to see NULL by phpDoc.

 But `private $_viewPath = null;`  is redundant.

Reply:
Removed default null values

Reply:
Oh sorry, removed phpDoc changes also. Do I add them, back?

Reply:
Yea. phpDoc was not bad.

Reply:
Done.